tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -218,5 +221,39 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            var logDaysLimit = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.LogFileRetentionDays) ?? "32";
+
+            if (int.TryParse(logDaysLimit, out var days) && days > 0)
+            {
+                var date = DateTime.Now.AddDays(-days);
+                var logFormats = new[]
+                  {
+                    "dotnet-tracer-*.log",
+                    "dotnet-native-loader-*.log",
+                    "DD-DotNet-Profiler-Native-*.log"
+                  };
+
+                try
+                {
+                    foreach (var logFormat in logFormats)
+                    {
+                        foreach (var logFile in Directory.EnumerateFiles(logsDirectory, logFormat))
+                        {
+                            if (File.GetLastWriteTime(logFile) < date)
+                            {
+                                File.Delete(logFile);
+                            }
+                        }
+                    }
+                }
+                catch
+                {
+                    // Abort on first catch when doing IO operation for performance reasons.
```

Comment:
It would be nice if we could do the cleanup after the logging system is available so we could log this exception. Not sure if it's worth it, though.

Reply:
Sounds good. Thanks!